src/smallfry.c

```diff
@@ -93,7 +93,7 @@ static double aae_factor(uint8_t *orig, uint8_t *cmp, int orig_stride,
     old = orig + 7 * orig_stride;
     new = cmp  + 7 * cmp_stride;
 
-    for (i = 7; i < height - 1; i += 8) {
+    for (i = 7; i < height - 2; i += 8) {
```

Comment:
Are there two!:
https://github.com/danielgtaylor/jpeg-archive/blob/e3deb06965f01b41427df0b2e884538c813326c5/src/smallfry.c#L75
https://github.com/danielgtaylor/jpeg-archive/blob/e3deb06965f01b41427df0b2e884538c813326c5/src/smallfry.c#L96

Reply:
Do you have a reproducer which crashes after this patch?

Reply:
See #109 .

Reply:
Do you have an image which crashes after this patch?

Reply:
I do not understand. Which picture?

Reply:
I'll post the link here as well: https://user-images.githubusercontent.com/3889750/66255383-0e68ec00-e783-11e9-8a05-34ea7900150e.jpg

Reply:
With this patch, that file compresses fine for me.

```$ ./jpeg-recompress -m smallfry <(curl https://user-images.githubusercontent.com/3889750/66255383-0e68ec00-e783-11e9-8a05-34ea7900150e.jpg) /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2802  100  2802    0     0  13471      0 --:--:-- --:--:-- --:--:-- 13471
Metadata size is 2kb
smallfry at q=67 (40 - 95): 97.887085
smallfry at q=81 (68 - 95): 98.664391
smallfry at q=88 (82 - 95): 95.794510
smallfry at q=92 (89 - 95): 111.058563
smallfry at q=90 (89 - 91): 115.722046
Final optimized smallfry at q=89: 107.025574
New size is 91% of original (saved 0 kb)
```

I also tried building with -fsanitize=address and it works fine.

If you have a file which you have observed does not work properly with smallfry with this patch applied, I would be happy to look at it.

Reply:
@jlebar say:
>If you have a file

Too big file (15000x10000).

Reply:
Hi, I think I am not communicating clearly.

I'm just a random person like you who uses this software.  I fixed a crasher bug that I found on a file of mine.  I think you are saying that you think there's still a bug.  Cool, I may be motivated to try to fix this bug!  But I will need your help in a specific way.

1. I need a JPEG file which causes a crash.  I need a specific JPEG file, not a description of a JPEG file.  That is, "too big file, 15000x15000" is not sufficient, because that is a description of a JPEG file and not a JPEG file itself.

2. I need someone to build jpeg-archive at HEAD (i.e. with the patch that I wrote) and verify that it crashes with the file from step 1 above.

3. I need an exact set of steps to reproduce the crash, i.e. the exact file which causes the crash, the exact command-line you ran, and the exact git revision you built at.

If you give me these things I may try to work on this issue.  If you can't, I cannot work on the issue and won't respond further on this thread.

Thanks.  :)

Reply:
@jlebar say:
>I think you are saying that you think there's still a bug.

Hmm. I am not a user of this software. I use the fork: https://github.com/ImageProcessing-ElectronicPublications/jpeg-recompress. But I'm interested in making the fork code as similar as possible to this software (or vice versa).